### ISSUE TITLE
Improve styling of dialogs to avoid unwanted layout shift when scaling up the application

### ISSUE THREAD

**Issue by AlbyIanna** [arduino/arduino-ide#1309]
We've noted that the layouts of some dialogs in the IDE2 become distorted (with buttons going offscreen), when scaling up the application.

We should add css rules to ensure dialog layout shifts that impair usability do not occur on scaling up the app. Generally speaking our dialogs should have a fixed header and footer, content between should not overflow and push things out of view. 

We should review all dialogs at different scales in the IDE2, we can group simpler dialogs that share rules into single PR, and use separate PRs for more complex dialogs (like the board / port selector).

Example of distorted board selector dialog when app is scaled up and the window width is reduced:
![Screenshot 2022-07-28 at 12.45.47.png](https://zube.io/files/arduino/126db35a47c749ed5a5e2283c0f956ec-screenshot-2022-07-28-at-12-45-47.png)

**Comment by InstantMuffin** [comment]
If this is not related, I'd like to open another issue specifically for this, but I have noticed throughout the release candidates that when you maximize the main window before it is fully loaded, the entire window content will be drawn using the initial dimensions using startup, leaving the rest blank.

Now as I am typing this and want to reproduce this, I just cannot do that successfully. I am using cdeb502.

**Comment by per1234** [comment]
@InstantMuffin that bug is being tracked here: https://github.com/arduino/arduino-ide/issues/1244

A fix was made for it in https://github.com/arduino/arduino-ide/pull/1255, before which I believe it was easier to reproduce. I produced it by chance yesterday, so I know it is still possible, but I did not reproduce it after that so I think either the time window for making the resize is very specific or else there are additional undetermined required conditions.

If you encounter it again, and especially if you find a way to reproduce it reliably, please comment on https://github.com/arduino/arduino-ide/issues/1244
